server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -2745,15 +2745,19 @@ void handle_b_fill(ReceivedMessage const & msg)
         return;
     }
 
+    int bufSamples = sc_factory->get_buffer_struct(buffer_index)->samples;
+
     while (it != end) {
         osc::int32 index = it->AsInt32(); ++it;
         verify_argument(it, end);
         osc::int32 samples = it->AsInt32(); ++it;
         verify_argument(it, end);
         float value = extract_float_argument(it++);
 
-        for (int i = 0; i != samples; ++i)
+        for (int i = 0; i != samples; ++i) {
             data[index] = value;
+            if(++index >= bufSamples) { break; }
```

Comment:
thanks! minor code style issues here: please move `++index` into its own line, add a space between `if (`, and move `break` to a separate line